pkg/monitors/monitor-service.go

```diff
@@ -12,10 +14,54 @@ type MonitorService interface {
 	GetByName(name string) (*models.Monitor, error)
 	Remove(m models.Monitor)
 	Setup(p config.Provider)
+	Equal(oldMonitor models.Monitor, newMonitor models.Monitor) bool
 }
 
 func CreateMonitorService(p *config.Provider) MonitorServiceProxy {
 	monitorService := (&MonitorServiceProxy{}).OfType(p.Name)
 	monitorService.Setup(*p)
 	return monitorService
 }
+
+func SetupMonitorServicesForProviders(providers []config.Provider) []MonitorServiceProxy {
+	if len(providers) < 1 {
+		log.Panic("Cannot Instantiate controller with no providers")
+	}
+
+	monitorServices := []MonitorServiceProxy{}
+
+	for index := 0; index < len(providers); index++ {
+		monitorServices = append(monitorServices, CreateMonitorService(&providers[index]))
+		log.Info("Configuration added for " + providers[index].Name)
+	}
+
+	return monitorServices
+}
+
+func SetupMonitorServicesForProvidersTest(providers []config.Provider) []MonitorServiceProxy {
+	if len(providers) < 1 {
+		log.Panic("Cannot Instantiate controller with no providers")
+	}
+	// TODO: Fix provider specific implementation and then add them to this list
+	allowedProviders := []string{"UptimeRobot", "StatusCake"}
+
+	monitorServices := []MonitorServiceProxy{}
+
+	for index := 0; index < len(providers); index++ {
+		if contains(allowedProviders, providers[index].Name) {
```

Comment:
we should log a message for the unsupported providers at least so the user knows why it doesn't work at this stage

Reply:
I'll add an additional log here although this is only applicable for test cases. Currently, only UptimeRobot and StatusCake are supported for reconciliation test cases

Reply:
Oh Ok I see that it's only for test cases